clang/test/Driver/darwin-header-search-libcxx.cpp

```diff
@@ -172,3 +172,20 @@
 // RUN:               --check-prefix=CHECK-LIBCXX-STDLIB-UNSPECIFIED %s
 // CHECK-LIBCXX-STDLIB-UNSPECIFIED: "-cc1"
 // CHECK-LIBCXX-STDLIB-UNSPECIFIED: "-internal-isystem" "[[SYSROOT]]/usr/include/c++/v1"
+
+// Reproduce the xPack use case; there must be no include here,
+// to select the executable folder.
+// RUN: rm -rf %t/xpacks
+// RUN: mkdir -pv %t/xpacks/.bin
+// RUN: ln -svf %clang %t/xpacks/.bin/clang
+// The build folders do not include this include; create it.
+// RUN: mkdir -pv $(dirname $(which %clang))/../include/c++/v1
```

Comment:
I think it would be better to create a new temporary folder with the appropriate structure, and copy the clang binary into it, then make the symlink and FileCheck the resulting cc1 paths.

Also, I don't think `ln` will work on Windows buildbots. You may need to put this test in its own file so you can add this at the top:

```
// REQUIRES: shell
// UNSUPPORTED: system-windows
```

Reply:
> `// UNSUPPORTED: system-windows`

The current file already has this definition at top. Isn't this enough?

> create a new temporary folder with the appropriate structure, and copy the clang binary into it

What if the binaries are built with absolute RPATH definitions?

Reply:
> Isn't this enough?

Sorry, missed that. Yes, it is.

> Is this solution used by other tests?

Not that I'm aware of. I don't know of any other tests that need Dir != InstalledDir either.

> What if the binary is built with weird RPATH definitions?

Good point.